### DISCUSSION TITLE
How to get a heart symbol filled with black color

### DISCUSSION THREAD

**Discussion by AjayDhimanELS** [KaTeX/KaTeX#2546] [Q&A] [answered]
1.  `\spadesuit` or `\spades` renders spade with filled black color.
2.  Similarly, `\clubs` or `\clubsuit` renders club with filled black color.
3.  Now `\hearts` or `\heartsuit` just renders heart with white filled bg.
4.  `\colorbox` treats this as a box/rectangle, thus hiding the shape obviously.

     [Please see this in new tab](https://katex.org/?data=%7B%22displayMode%22%3Atrue%2C%22leqno%22%3Afalse%2C%22fleqn%22%3Afalse%2C%22throwOnError%22%3Atrue%2C%22errorColor%22%3A%22%23cc0000%22%2C%22strict%22%3A%22warn%22%2C%22output%22%3A%22htmlAndMathml%22%2C%22trust%22%3Afalse%2C%22macros%22%3A%7B%22%5C%5Cf%22%3A%22%231f(%232)%22%7D%2C%22code%22%3A%22%5C%5Cspades%5Cn%5C%5C%5C%5C%5Cn%5C%5Cspadesuit%5Cn%5C%5C%5C%5C%5Cn%5C%5Cclubs%5Cn%5C%5C%5C%5C%5Cn%5C%5Cclubsuit%5Cn%5C%5C%5C%5C%5Cn%5C%5Chearts%5Cn%5C%5C%5C%5C%5Cn%5C%5Cheartsuit%22%7D)

5. I get it that club and spade are black and the diamonds and heart are red in a deck, but I want heart symbol in black as well. 
6. Okay, can use `\blacklozenge` for diamond. But no clue for heart.
How can I get that? Thanks.

Asked on Stack Overflow too : https://stackoverflow.com/questions/64043621/how-to-get-a-heart-filled-of-black-color-in-katex

**Comment by ronkok** [comment, marked as answer]
KaTeX `\hearts` and `\heartsuit` render a heart outline, ♡, U+2661, , so they do not help you. The Unicode character for a **filled** heart is ♥, U+2665. That glyph is not in the KaTeX fonts, but you can still use it and KaTeX will substitute the glyph from the Times New Roman system font.

KaTeX emulates LaTeX, and this is not a LaTeX function, so KaTeX will write at least one error to the console (two if in strict mode). If you don't mind a console error, then no worries.  KaTeX does render the heart. Just copy ♥ and paste it into your work.

**Reply by AjayDhimanELS** [reply]
Actually, I also thought of this. But 🖤 will not work

**Reply by AjayDhimanELS** [reply]
Hmm. Until there's any further update to this. This is what we'll have to work with. Thanks for this.

**Comment by edemaine** [comment]
Looking at [the LaTeX symbols list](http://ctan.math.utah.edu/ctan/tex-archive/info/symbols/comprehensive/symbols-a4.pdf), there are a few packages that offer `\varheartsuit` as a filled variant of the heart. Something we could eventually add to KaTeX's fonts...
